### ISSUE TITLE
Make AST fully inspectable

### ISSUE THREAD

**Issue by liferooter** [nushell/nushell#11687]
### Related problem

I've faced this issue when I've tried to implement function that shows man page for the command under the cursor, just like `__fish_man_page` in Fish. I tried to use `commandline` in pair with `ast`, but got stuck. The problem is that `ast` command just dumps internal representation of AST which doesn't easy to use without its Rust API available and also doesn't contain all the information. For example, closures are represented by some internal ID which is not accessible from Nu scripts, so some of tree is lost.

### Describe the solution you'd like

I suggest to create some easy-to-use and fully informative AST format for introspection and add some convenient commands to make working with AST easier (e.g. some commands to get subexpressions of the expression).

### Describe alternatives you've considered

Probably another alternative is to rework Nu's internal representation of AST to make it usable for introspection from outside. I don't know which of the ways is harder, but it seems for me that it's easier to create some extra types representing AST and convertable from internal AST representation than to fully rework internal AST representation for introspection uses. 

### Additional context and details

Here is some list of currently unsolvable or hard-to-solve problems with AST:

- `Expr::Var`, `Expr::VarDecl`, `Expr::RowCondition`, `Expr::Subexpression`, `Expr::Block`, `Expr::Closure`, `Expr::Overlay`, `Expr:: and partially `Expr::MatchBlock` and `Expr::Signature` are unintrospectable due to use of internal IDs.
- `serde`'s JSON representation of Rust structures is hard to use from Nu, leading to huge `match`es with record patterns instead of simple string values (as if there would be `type` field and fields instead of list for tuple types and enum variants)
- Nu's global span indexing is ugly and inconvenient for external use
- There's no good way to get expression from a pipeline element or list of subexpressions from an expression without rewritting huge amount of Nu's code from Rust to Nu

Additionally it would be cool to have some convenient way to get the stack of concentric AST nodes around the cursor. It would make writting some

**Comment by fdncred** [comment]
There's also `view files`, `view source`, and `view span` that can help with this. you can find all of these type of commands with this nushell statement.
```nushell
help commands | where category =~ debug
```
I'm not saying we have everything you've mentioned, just trying to shine a light on what we have already.

**Comment by liferooter** [comment]
Yes, I know. That's why global span indexing isn't impossible to use, but still inconvenient.
